example/functions/getComposedResource/composition.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-function-get-composed-resource
+spec:
+  compositeTypeRef:
+    apiVersion: example.crossplane.io/v1beta1
+    kind: XR
+  mode: Pipeline
+  pipeline:
+    - step: render-templates
+      functionRef:
+        name: function-go-templating
+      input:
+        apiVersion: gotemplating.fn.crossplane.io/v1beta1
+        kind: GoTemplate
+        source: Inline
+        inline:
+          template: |
+            ---
+            # Create an initial composed resource for which we will retrieve a value
+            apiVersion: dbforpostgresql.azure.upbound.io/v1beta1
+            kind: FlexibleServer
+            metadata:
+              annotations:
+                {{ setResourceNameAnnotation "flexserver" }}
+                gotemplating.fn.crossplane.io/ready: "False"
+            spec:
+              forProvider:
+                storageMb: 32768
+              providerConfigRef:
+                name: my-provider-cfg
+            ---
+            # Use getComposedResource to retrieve the observed resource named "flexServer"
+            {{ $flexServer := getComposedResource . "flexServer" }}
```

Comment:
Looks like the resource was named `flexserver` all lowercase instead of `flexServer`.

Reply:
fixed :)